presto-spi/src/main/java/io/prestosql/spi/type/LongTimestampWithTimeZone.java

```diff
@@ -66,4 +69,34 @@ public short getTimeZoneKey()
     {
         return timeZoneKey;
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LongTimestampWithTimeZone that = (LongTimestampWithTimeZone) o;
+        return epochMillis == that.epochMillis &&
+                picosOfMilli == that.picosOfMilli;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(epochMillis, picosOfMilli);
+    }
+
+    @Override
+    public int compareTo(LongTimestampWithTimeZone other)
+    {
+        int value = Long.compare(epochMillis, other.epochMillis);
+        if (value != 0) {
+            return value;
+        }
+        return Integer.compareUnsigned(picosOfMilli, other.picosOfMilli);
```

Comment:
`Integer.compare`

Reply:
This matches the compare implementation in the Type class.